### PR TITLE
fix: solves display bug of category chevron buttons

### LINKED ISSUE #628: Bug (with solution PR): Video Category's Chevron Icons update late based on click event
### Is there an existing issue for this?

- [X] I have searched the existing issues

## Note: I have also issued a Pull Request based on this Bug, please consider it after inspecting this bug report. Thanks!

The Category scroll bar does not update (hide/show) its `ChevronRightOutline` or `ChevronLeftOutline` as scrolling is done by updating the `useRef`.

Here's how the UI Normally looks:
Issue

Here's how it looks after we click on `>` ONCE, the Red box indicates that the `<` icon should already be visible now:
Issue1

Here's how it looks RIGHT AFTER we reach the end of the scroll list, the red box indicates that the `>` icon should already disappear now:
Issue2

### Expected Behavior

The expected behavior **(as also shown as a solution in my PR)** is that the `<` and `>` icons are displayed correctly and not after one more (unnecessary) click event to be done to update/re-render the UI.

## This is how the proposed solution in my PR looks like:
**The URL shows the difference between the current behavior (lenstube) and the expected behavior (localhost)**

https://user-images.githubusercontent.com/20269286/219801637-929afbb0-0477-4449-b05b-8a4f7b84fa75.mp4

### Steps To Reproduce

The original website currently shows the same behavior, kindly click on the `>` and `<` of the category section to see the bug.

### What platform(s) does this occur on?

Web

### What browser(s) does this occur on?

Chrome

### Anything else?

_No response_

### PR DESCRIPTION
**Pull request by abhiram11**
## Description

Closes #628 

- [x] I read the [contributing docs](CONTRIBUTING.md) (if this is your first contribution)

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented on my code, particularly in hard-to-understand areas
- [x] My changes generate no new warnings

Note: Kindly check the Bug Report #628 where I have elaborated on the issue in more detail.

This PR solves the display bug where the `<` and `>` category buttons require an extra event to be called/clicked in order to render on the page as required by the logic.

## This is how the proposed solution in my PR looks like:
**The URL shows the difference between the current behavior (lenstube) and the expected behavior (localhost)**

https://user-images.githubusercontent.com/20269286/219801637-929afbb0-0477-4449-b05b-8a4f7b84fa75.mp4

### P.S. I am new to open-source contributions, I am all ears to feedback! Furthermore, we can also refactor the filename `utils/hooks/useHorizantalScroll` and correct the spelling to `Horizontal`. ^^

### CODE REVIEW

File: apps/web/src/components/Common/CategoryFilters.tsx [hunk 1]
```diff
@@ -38,14 +38,31 @@ const CategoryFilters = () => {
 
   const slide = (shift: number) => {
     Analytics.track(TRACK.CLICK_CATEGORIES_SCROLL_BUTTON)
+
     if (scrollRef.current) {
+      /**
```
Comment (sasicodes) [inline apps/web/src/components/Common/CategoryFilters.tsx:43]: Great work!

Thanks for the comments to convey. We can clean up the comments 👍🏻

File: apps/web/src/components/Common/CategoryFilters.tsx [hunk 2]
```diff
@@ -38,14 +38,31 @@ const CategoryFilters = () => {
...
+       * 
+      I observed that since updating the Ref doesn't re-render the UI,
+      the scrollX/ChevronLeftOutline was still invisible after the FIRST CLICK on scrollEnd/ChevronRightOutline
+      Also, this calculation:
+              scrollRef.current.scrollLeft += shift
+      did not update the value of scrollRef.current.scrollLeft on the go
+       */
+
+      // That's why I used a new variable to handle the calculation of "shift" effectively
+      let scrolled = scrollRef.current.scrollLeft + shift
```
Comment (sasicodes) [inline apps/web/src/components/Common/CategoryFilters.tsx:55]: can be `const` if we are not reassigning the value 🙇🏻

**Review by sasicodes — CHANGES_REQUESTED** [review]

**Comment by sasicodes** [comment]
@abhiram11 We can rename typo in the the file name to correct spelling 👍🏻

File: apps/web/src/components/Common/CategoryFilters.tsx (hunk 2 above)
Comment (abhiram11) [reply apps/web/src/components/Common/CategoryFilters.tsx:55]: Sure! Let me update that, I overlooked the type after the whole debugging I guess.

File: apps/web/src/components/Common/CategoryFilters.tsx (hunk 1 above)
Comment (abhiram11) [reply apps/web/src/components/Common/CategoryFilters.tsx:43]: Thanks for the feedback, I will remove the comments and set the variable type to `const` in the next PR.

Should I change the filename in this one itself or make a new PR for it?

Comment (sasicodes) [reply apps/web/src/components/Common/CategoryFilters.tsx:43]: Update all comments in this same PR!

**Comment by abhiram11** [comment]
I have pushed a new commit with the following updates:
1. Changed `let` to `const`
2. Removed Comments
3. Updated filename to `useHorizontalScroll` and modified wherever it is called.

P.S. I am new to open source contribution and git, so if there are any issues with my PR or git calls, please provide feedback for the same!

**Comment by sasicodes** [comment]
> I have pushed a new commit with the following updates:
> 
> 1. Changed `let` to `const`
> 2. Removed Comments
> 3. Updated filename to `useHorizontalScroll` and modified wherever it is called.
> 
> P.S. I am new to open source contribution and git, so if there are any issues with my PR or git calls, please provide feedback for the same!

You did all great. Thank you for your contribution. 😊

**Comment by abhiram11** [comment]
Thank you for the feedback! ^^ Feel free to let me know if there are any other issues that I can help with, especially in Javascript, Python, and Frontend.

Have a great weekend!

**Review by sasicodes — APPROVED** [review]
Dog LGTM Keyboard

**Comment by sasicodes** [comment]
> 

You can always pick from issues or suggest feature requests (roadmap.lenstube.xyz)

**Comment by sasicodes** [comment]
@abhiram11 ☝🏻

**Comment by abhiram11** [comment]
This is great! Thanks for the recognition!! 😊

**Pull request merged**